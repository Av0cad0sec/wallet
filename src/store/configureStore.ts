import { routerMiddleware } from 'connected-react-router'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { createInjectorsEnhancer } from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'

import { createReducer, history } from './reducers'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.message,
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })
  const { run: runSaga } = sagaMiddleware

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)]

  const enhancers = [
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ]

  const store = configureStore({
    reducer: createReducer(),
    middleware: [...getDefaultMiddleware(), ...middlewares],
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production' || process.env.PUBLIC_URL.length > 0,
    enhancers,
    preloadedState: state,
  })

  return store
}
