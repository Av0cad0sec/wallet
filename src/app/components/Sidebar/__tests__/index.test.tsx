import { render } from '@testing-library/react'
import { ConnectedRouter } from 'connected-react-router'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'

import { Sidebar } from '..'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

const renderComponent = store => {
  const history = createBrowserHistory()
  return render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Sidebar />
      </ConnectedRouter>
    </Provider>,
  )
}

describe('<Sidebar />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
    jest.resetAllMocks()
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
