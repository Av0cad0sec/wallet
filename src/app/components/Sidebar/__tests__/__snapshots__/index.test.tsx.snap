// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should match snapshot 1`] = `
.c11 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c11 g {
  fill: inherit;
  stroke: inherit;
}

.c11 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c11 *[stroke*="#"],
.c11 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c11 *[fill-rule],
.c11 *[FILL-RULE],
.c11 *[fill*="#"],
.c11 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: component-sidebar;
  border-inline-end: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
  width: 220px;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 0px;
  padding-right: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 24px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: url(https://avatars.githubusercontent.com/u/52803776?s=200&v=4) no-repeat center center;
  background-size: cover;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 48px;
  width: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 100%;
  overflow: hidden;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: background-oasis-blue;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 4px;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding: 12px;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c5 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c12 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c16:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus > circle,
.c16:focus > ellipse,
.c16:focus > line,
.c16:focus > path,
.c16:focus > polygon,
.c16:focus > polyline,
.c16:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus::-moz-focus-inner {
  border: 0;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
}

@media only screen and (max-width:768px) {
  .c0 {
    border-inline-end: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-left: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-right: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding-left: 6px;
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding-top: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c17 {
    border: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c18 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c19 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    height: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 StyledAvatar-sc-1suyamb-1"
    />
    <div
      class="c3"
    />
    <span
      class="c4"
    >
      Oasis Wallet
    </span>
  </div>
  <div
    class="c5"
  />
  <div
    class="c6"
  >
    <nav
      class="c7"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <div
          class="c8"
        >
          <button
            aria-label="menu.home"
            class="c9"
            data-testid="nav-home"
            type="button"
          >
            <div
              class="c10"
            >
              <svg
                aria-label="Home"
                class="c11"
                viewBox="0 0 24 24"
              >
                <path
                  d="M1,11 L12,2 L23,11 M15,23 L15,15 L15,15 L9,15 L9,23 M4,23 L4,9 M20,23 L20,9"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
              <div
                class="c12"
              />
              menu.home
            </div>
          </button>
        </div>
      </a>
      <div
        class="c5"
      />
      <div
        class="c5"
      />
    </nav>
  </div>
  <div
    class="c5"
  />
  <nav
    class="c13"
  >
    <div
      class="c14"
    >
      <button
        aria-label="theme.darkMode"
        class="c9"
        type="button"
      >
        <div
          class="c10"
        >
          <svg
            aria-label="Moon"
            class="c11"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.2494 18.7793L18.7813 19.6261C19.1196 19.4136 19.299 19.0202 19.2375 18.6255C19.1761 18.2308 18.8856 17.9105 18.4988 17.8109L18.2494 18.7793ZM18.2494 5.22068L18.4988 6.18909C18.8856 6.08949 19.1761 5.76921 19.2375 5.37449C19.299 4.97978 19.1196 4.58636 18.7813 4.37388L18.2494 5.22068ZM14 21C15.755 21 17.3954 20.4967 18.7813 19.6261L17.7175 17.9325C16.641 18.6088 15.3677 19 14 19V21ZM5 12C5 16.9706 9.02944 21 14 21V19C10.134 19 7 15.866 7 12H5ZM14 3C9.02944 3 5 7.02944 5 12H7C7 8.13401 10.134 5 14 5V3ZM18.7813 4.37388C17.3954 3.50331 15.755 3 14 3V5C15.3677 5 16.641 5.39124 17.7175 6.06748L18.7813 4.37388ZM18.0001 4.25226C14.5502 5.14052 12 8.27118 12 12H14C14 9.20579 15.911 6.85538 18.4988 6.18909L18.0001 4.25226ZM12 12C12 15.7288 14.5502 18.8595 18.0001 19.7477L18.4988 17.8109C15.911 17.1446 14 14.7942 14 12H12Z"
              fill="#000"
            />
          </svg>
          <div
            class="c12"
          />
          theme.darkMode
        </div>
      </button>
    </div>
    <div
      class="c5"
    />
    <div
      class="c5"
    />
    <div
      class="c15"
    >
      <button
        aria-label="Open Menu"
        class="c16"
        type="button"
      >
        <div
          class="c17"
        >
          <div
            class="c18"
          >
            <svg
              aria-label="Language"
              class="c11"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,23 C18.0751322,23 23,18.0751322 23,12 C23,5.92486775 18.0751322,1 12,1 C5.92486775,1 1,5.92486775 1,12 C1,18.0751322 5.92486775,23 12,23 Z M12,23 C15,23 16,18 16,12 C16,6 15,1 12,1 C9,1 8,6 8,12 C8,18 9,23 12,23 Z M2,16 L22,16 M2,8 L22,8"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="c19"
          >
            <span
              class="c4"
            >
              Language
            </span>
          </div>
          <div
            class="c18"
          >
            <svg
              aria-label="FormDown"
              class="c11"
              viewBox="0 0 24 24"
            >
              <polyline
                fill="none"
                points="18 9 12 15 6 9"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
  </nav>
</div>
`;
